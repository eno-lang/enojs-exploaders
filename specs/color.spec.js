const eno = require('enojs');
const loaders = require('../loaders.js');

describe('color', () => {

  let document;

  describe('#abcdef', () => {
    it("returns '#abcdef'", () => {
      document = eno.parse('color: #abcdef');
      expect(document.field('color', loaders.color)).toEqual('#abcdef');
    });
  });

  describe('#ABCDEF', () => {
    it("returns '#ABCDEF'", () => {
      document = eno.parse('color: #ABCDEF');
      expect(document.field('color', loaders.color)).toEqual('#ABCDEF');
    });
  });

  describe('#012345', () => {
    it("returns '#012345'", () => {
      document = eno.parse('color: #012345');
      expect(document.field('color', loaders.color)).toEqual('#012345');
    });
  });

  describe('#678', () => {
    it("returns '#678'", () => {
      document = eno.parse('color: #678');
      expect(document.field('color', loaders.color)).toEqual('#678');
    });
  });

  describe('#89a', () => {
    it("returns '#89a'", () => {
      document = eno.parse('color: #89a');
      expect(document.field('color', loaders.color)).toEqual('#89a');
    });
  });

  describe('#ab', () => {
    it('throws an error', () => {
      document = eno.parse('color: #ab');
      expect(() => document.field('color', loaders.color)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('#abcd', () => {
    it('throws an error', () => {
      document = eno.parse('color: #abcd');
      expect(() => document.field('color', loaders.color)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('#abcde', () => {
    it('throws an error', () => {
      document = eno.parse('color: #abcde');
      expect(() => document.field('color', loaders.color)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('#bcdefg', () => {
    it('throws an error', () => {
      document = eno.parse('color: #bcdefg');
      expect(() => document.field('color', loaders.color)).toThrowErrorMatchingSnapshot();
    });
  });
});
