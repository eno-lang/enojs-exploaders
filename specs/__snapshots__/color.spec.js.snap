// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`color #ab throws an error 1`] = `
"'color' must be a color (#RRGGBB or #RGB).

   Line | Content
 >    1 | color: #ab
"
`;

exports[`color #abcd throws an error 1`] = `
"'color' must be a color (#RRGGBB or #RGB).

   Line | Content
 >    1 | color: #abcd
"
`;

exports[`color #abcde throws an error 1`] = `
"'color' must be a color (#RRGGBB or #RGB).

   Line | Content
 >    1 | color: #abcde
"
`;

exports[`color #bcdefg throws an error 1`] = `
"'color' must be a color (#RRGGBB or #RGB).

   Line | Content
 >    1 | color: #bcdefg
"
`;
